sktime/utils/_testing/estimator_checks.py

```diff
@@ -97,6 +98,96 @@ def yield_estimator_checks(exclude=None):
         yield check
 
 
+def tuple_equals(x, y):
```

Comment:
Regarding tests for tests, we may want one for this function

Reply:
which folder do tests for tests go in?

Reply:
I would move the function to `sktime/utils/_testing` and then add tests in `sktime/utils/_testing/tests`

Reply:
I moved the utility to the indicated folder, but *didn't* delete the function. Instead, I refactored it into a single `deep_equals` which also allows to recurse over lists/tuples/dicts of lists/tuples/dicts.

Why: in the panel setting, in the distance module, and in tuning (parameter sets), we have cases where arguments are lists or dicts, so for the comparison of the entire argument set we get a tuple of lists or tuple of dict (of numpy etc).